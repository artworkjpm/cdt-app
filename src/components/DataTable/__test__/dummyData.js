export const dummyDataDesigns = [
	{
		id: 20,
		name: "20th Design",
		courses: 111,
		wales: 333,
		last_updated: "12/04/2021",
		by: "WD",
	},
];

export const dummyDataSetouts = [
	{
		courses: 200,
		id: 1,
		last_updated: "24/03/2021",
		machine_name: "RSJ_4_1",
		machine_width: 130,
		name: "1st Setout",
	},
];
