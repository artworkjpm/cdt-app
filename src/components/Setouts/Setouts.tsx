import axios from "axios";
import { useEffect, useState } from "react";
import InfiniteScroll from "react-infinite-scroll-component";
import { useDispatch, useSelector } from "react-redux";
import { SetoutsModel } from "../../models/interfaces";
import { fetchSetouts, updateAmount } from "../../redux/actions/setoutActions";
import DataTable from "../DataTable/DataTable";

export default function Setouts() {
	const dispatch = useDispatch();
	const data = useSelector((state: { setOutsReducer: { setouts: [SetoutsModel] } }) => state.setOutsReducer.setouts);
	const until = useSelector((state: { setOutsReducer: { until: number } }) => state.setOutsReducer.until);
	const [openDialog, setOpenDialog] = useState(false);
	useEffect(() => {
		!data.length && dispatch(fetchSetouts(10));
	}, [dispatch, data]);

	const tableHeaders = ["Name", "Machine_Name", "Machine_Width", "Courses", "Last_Updated"];
	const editableFields = ["name", "machine_name", "machine_width", "courses"];

	function handleScroll() {
		dispatch(updateAmount(until + 10));
		dispatch(fetchSetouts(until + 10));
	}

	const onSubmit = (editedItem: any) => {
		axios
			.put(`http://localhost:5000/setouts/${editedItem.id}`, editedItem)
			.then(() => {
				dispatch(fetchSetouts(until));
				setOpenDialog(false);
			})
			.catch((error) => {
				console.error(error);
				setOpenDialog(false);
			});
	};

	function handleOpen() {
		setOpenDialog(true);
	}

	function handleClose() {
		setOpenDialog(false);
	}

	return (
		<div className="scroll-div" onScroll={handleScroll}>
			<InfiniteScroll dataLength={data.length} next={handleScroll} hasMore={true} loader="">
				<DataTable handleOpen={handleOpen} handleClose={handleClose} openDialog={openDialog} editTitle="Setout" tableHeaders={tableHeaders} data={data} editableFields={editableFields} onEditSubmit={onSubmit}></DataTable>
			</InfiniteScroll>
		</div>
	);
}
