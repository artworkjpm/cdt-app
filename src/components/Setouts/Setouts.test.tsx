import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { dummyDataSetouts } from "../DataTable/__test__/dummyData";
import Setouts from "./Setouts";
const mockStore = configureMockStore([thunk]);

describe("<Setouts>", () => {
	it("should render a startup component if startup is not complete", () => {
		const store = mockStore({
			setOutsReducer: { setouts: dummyDataSetouts },
		});
		const wrapper = mount(
			<Provider store={store}>
				<Setouts />
			</Provider>
		);
		expect(wrapper.debug()).toMatchSnapshot();
	});
});
