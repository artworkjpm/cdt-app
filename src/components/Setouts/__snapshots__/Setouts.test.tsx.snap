// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Setouts> should render a startup component if startup is not complete 1`] = `
"<Provider store={{...}}>
  <Setouts>
    <div className=\\"scroll-div\\" onScroll={[Function: handleScroll]}>
      <InfiniteScroll dataLength={1} next={[Function: handleScroll]} hasMore={true} loader=\\"\\">
        <div style={{...}} className=\\"infinite-scroll-component__outerdiv\\">
          <div className=\\"infinite-scroll-component \\" style={{...}}>
            <DataTable handleOpen={[Function: handleOpen]} handleClose={[Function: handleClose]} openDialog={false} editTitle=\\"Setout\\" tableHeaders={{...}} data={{...}} editableFields={{...}} onEditSubmit={[Function: onSubmit]}>
              <div id=\\"datatable\\">
                <table>
                  <thead>
                    <tr>
                      <th>
                        Name
                      </th>
                      <th>
                        Machine Name
                      </th>
                      <th>
                        Machine Width
                      </th>
                      <th>
                        Courses
                      </th>
                      <th>
                        Last Updated
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr onClick={[Function: onClick]}>
                      <td>
                        1st Setout
                      </td>
                      <td>
                        RSJ_4_1
                      </td>
                      <td>
                        130
                      </td>
                      <td>
                        200
                      </td>
                      <td>
                        24/03/2021
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </DataTable>
          </div>
        </div>
      </InfiniteScroll>
    </div>
  </Setouts>
</Provider>"
`;
